resource/config_map.go

```diff
@@ -0,0 +1,198 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type ConfigMapConfig struct {
+	// DesiredFn is function returning a desired ConfigMap object for the
```

Comment:
Is `Fn` a golang convention? We used `func` so far.

Reply:
I think `Fn` is commonly used in functional languages, but don't remember seeing it in stdlib (e.g. `sort` and `heap` don't use notion of `Fn` or `Func` but e.g. `less func(i, j int)`). As we use `Func` elsewhere, I'd also vote for that.

Reply:
I just gave an example from stdlib. Or you understand stdlib to be something else?
For me it's a minor. I can rename the suffix to `Func` if that's confusion. I based this on the `filepath.Walk` stuff.